sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/permission/put/RestPutPermissionAction.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.permission.put;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.identity.DefaultObjectMapper;
+import org.opensearch.identity.rest.IdentityRestConstants;
+import org.opensearch.identity.utils.ErrorType;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.opensearch.identity.utils.RoutesHelper.addRoutesPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Rest action for adding a permission to the permission store
+ */
+public class RestPutPermissionAction extends BaseRestHandler {
+
+    /**
+     * @return a string of the action name  -- "_identity/api/permissions/put
+     */
+    @Override
+    public String getName() {
+        // permission_action_put
+        return IdentityRestConstants.PERMISSION_ACTION_PREFIX + "_put";
+    }
+
+    /**
+     * Rest request handler for granting a permission to a subject
+     * @param request the request to execute
+     * @param client  client for executing actions on the local node
+     * @return the action to be executed See {#handleRequest(RestRequest, RestChannel, NodeClient) for more}
+     * @throws IOException if errors encountered when parsing from XContent
+     *
+     * ````
+     * Sample Request:
+     * curl -XPUT http://new-user:password@localhost:9200/_identity/api/permissions/second_user/put --data '{ "permission" : "my_permission" }' -H"Content-type: application/json"
+     *
+     *
+     * Sample Response
+     *
+     * {
+     *   "permissions": [
+     *     {
+     *       "successful": true,
+     *       "permission added": "my_permission",
```

Comment:
Keys in a response should not contain spaces. 

IMO I think the response should be concise like `grant successful`. The username is not required in the response if its one of the params of the API.

Reply:
I was trying to keep the same formatting as the User API responses. I also think it may be helpful to have the username field present since a valid request could have your own username in the front of the URI as part of basic auth and then a second username later as part of the field. This makes it very clear in the response who is getting what.